rust-symcrypt/src/lib.rs

```diff
@@ -15,23 +15,50 @@ pub mod rsa;
 fn symcrypt_init() {
     // Subsequent calls to `symcrypt_init()` after the first will not be invoked per .call_once docs https://doc.rust-lang.org/std/sync/struct.Once.html
     static INIT: Once = Once::new();
+
+    // `symcrypt_init` calls `SymCryptModuleInit` or `SymCryptInit` depending on the feature flag
+    // We have also set feature flags on the bindings themselves to only expose the functions we need.
+    // This is to try and eliminate footguns like calling SymCryptModuleInit on a statically linked module.
     unsafe {
         // SAFETY: FFI calls, blocking from being run again.
+
+        #[cfg(feature = "dynamic")]
         INIT.call_once(|| {
             symcrypt_sys::SymCryptModuleInit(
                 symcrypt_sys::SYMCRYPT_CODE_VERSION_API,
                 symcrypt_sys::SYMCRYPT_CODE_VERSION_MINOR,
             )
         });
+
+        #[cfg(not(feature = "dynamic"))]
+        INIT.call_once(|| {
+            symcrypt_sys::SymCryptInit();
+        });
+        // call symcrypt Init, only expose this for static linking. Add a regex to add a cfg feature to the bindings for this.
     }
 }
 
 /// Takes in a a buffer called `buff` and fills it with random bytes. This function cannot fail.
```

Comment:
>This function cannot fail. [](http://example.com/codeflow?start=70&length=26)

I think it would be more accurate to say that it is never expected to fail, but failure (due to OS dependencies) will crash the application. There is no recoverable failure mode.